// @azure/arm-costmanagement/dist-esm/src/operations/generateDetailedCostReport.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { createHttpPoller } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl";
/** Class containing GenerateDetailedCostReport operations. */
export class GenerateDetailedCostReportImpl {
    /**
     * Initialize a new instance of the class GenerateDetailedCostReport class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Generates the detailed cost report for provided date range, billing period(only enterprise
     * customers) or Invoice ID asynchronously at a certain scope. Call returns a 202 with header
     * Azure-Consumption-AsyncOperation providing a link to the operation created. A call on the operation
     * will provide the status and if the operation is completed the blob file where generated detailed
     * cost report is being stored.
     * @param scope The ARM Resource ID for subscription, resource group, billing account, or other billing
     *              scopes. For details, see https://aka.ms/costmgmt/scopes.
     * @param parameters Parameters supplied to the Create detailed cost report operation.
     * @param options The options parameters.
     */
    beginCreateOperation(scope, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: { scope, parameters, options },
                spec: createOperationOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Generates the detailed cost report for provided date range, billing period(only enterprise
     * customers) or Invoice ID asynchronously at a certain scope. Call returns a 202 with header
     * Azure-Consumption-AsyncOperation providing a link to the operation created. A call on the operation
     * will provide the status and if the operation is completed the blob file where generated detailed
     * cost report is being stored.
     * @param scope The ARM Resource ID for subscription, resource group, billing account, or other billing
     *              scopes. For details, see https://aka.ms/costmgmt/scopes.
     * @param parameters Parameters supplied to the Create detailed cost report operation.
     * @param options The options parameters.
     */
    beginCreateOperationAndWait(scope, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreateOperation(scope, parameters, options);
            return poller.pollUntilDone();
        });
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOperationOperationSpec = {
    path: "/{scope}/providers/Microsoft.CostManagement/generateDetailedCostReport",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        201: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        202: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        204: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        default: {
            bodyMapper: Mappers.GenerateDetailedCostReportErrorResponse
        }
    },
    requestBody: Parameters.parameters6,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.scope1],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
