// @azure/arm-costmanagement/dist-esm/samples-dev/exportsGetExecutionHistorySample.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { CostManagementClient } from "@azure/arm-costmanagement";
import { DefaultAzureCredential } from "@azure/identity";
import * as dotenv from "dotenv";
dotenv.config();
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetByBillingAccount.json
 */
function exportRunHistoryGetByBillingAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/123456";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetByDepartment.json
 */
function exportRunHistoryGetByDepartment() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/12/departments/1234";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetByEnrollmentAccount.json
 */
function exportRunHistoryGetByEnrollmentAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/100/enrollmentAccounts/456";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetByManagementGroup.json
 */
function exportRunHistoryGetByManagementGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Management/managementGroups/TestMG";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetByResourceGroup.json
 */
function exportRunHistoryGetByResourceGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/MYDEVTESTRG";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to get the run history of an export for the defined scope and export name.
 *
 * @summary The operation to get the run history of an export for the defined scope and export name.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportRunHistoryGetBySubscription.json
 */
function exportRunHistoryGetBySubscription() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.getExecutionHistory(scope, exportName);
        console.log(result);
    });
}
function main() {
    return __awaiter(this, void 0, void 0, function* () {
        exportRunHistoryGetByBillingAccount();
        exportRunHistoryGetByDepartment();
        exportRunHistoryGetByEnrollmentAccount();
        exportRunHistoryGetByManagementGroup();
        exportRunHistoryGetByResourceGroup();
        exportRunHistoryGetBySubscription();
    });
}
main().catch(console.error);
