// @azure/arm-costmanagement/dist-esm/samples-dev/exportsListSample.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { CostManagementClient } from "@azure/arm-costmanagement";
import { DefaultAzureCredential } from "@azure/identity";
import * as dotenv from "dotenv";
dotenv.config();
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetByBillingAccount.json
 */
function exportsGetByBillingAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/123456";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetByDepartment.json
 */
function exportsGetByDepartment() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/12/departments/123";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetByEnrollmentAccount.json
 */
function exportsGetByEnrollmentAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/100/enrollmentAccounts/456";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetByManagementGroup.json
 */
function exportsGetByManagementGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Management/managementGroups/TestMG";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetByResourceGroup.json
 */
function exportsGetByResourceGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/MYDEVTESTRG";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to list all exports at the given scope.
 *
 * @summary The operation to list all exports at the given scope.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportsGetBySubscription.json
 */
function exportsGetBySubscription() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.list(scope);
        console.log(result);
    });
}
function main() {
    return __awaiter(this, void 0, void 0, function* () {
        exportsGetByBillingAccount();
        exportsGetByDepartment();
        exportsGetByEnrollmentAccount();
        exportsGetByManagementGroup();
        exportsGetByResourceGroup();
        exportsGetBySubscription();
    });
}
main().catch(console.error);
