// @azure/arm-costmanagement/dist-esm/samples-dev/exportsDeleteSample.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { CostManagementClient } from "@azure/arm-costmanagement";
import { DefaultAzureCredential } from "@azure/identity";
import * as dotenv from "dotenv";
dotenv.config();
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteByBillingAccount.json
 */
function exportDeleteByBillingAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/123456";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteByDepartment.json
 */
function exportDeleteByDepartment() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/12/departments/1234";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteByEnrollmentAccount.json
 */
function exportDeleteByEnrollmentAccount() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Billing/billingAccounts/100/enrollmentAccounts/456";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteByManagementGroup.json
 */
function exportDeleteByManagementGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "providers/Microsoft.Management/managementGroups/TestMG";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteByResourceGroup.json
 */
function exportDeleteByResourceGroup() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/MYDEVTESTRG";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to The operation to delete a export.
 *
 * @summary The operation to delete a export.
 * x-ms-original-file: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/examples/ExportDeleteBySubscription.json
 */
function exportDeleteBySubscription() {
    return __awaiter(this, void 0, void 0, function* () {
        const scope = "subscriptions/00000000-0000-0000-0000-000000000000";
        const exportName = "TestExport";
        const credential = new DefaultAzureCredential();
        const client = new CostManagementClient(credential);
        const result = yield client.exports.delete(scope, exportName);
        console.log(result);
    });
}
function main() {
    return __awaiter(this, void 0, void 0, function* () {
        exportDeleteByBillingAccount();
        exportDeleteByDepartment();
        exportDeleteByEnrollmentAccount();
        exportDeleteByManagementGroup();
        exportDeleteByResourceGroup();
        exportDeleteBySubscription();
    });
}
main().catch(console.error);
