// @azure/arm-costmanagement/dist-esm/src/costManagementClient.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { OperationsImpl, ViewsImpl, AlertsImpl, ForecastImpl, DimensionsImpl, QueryImpl, GenerateReservationDetailsReportImpl, ExportsImpl, GenerateCostDetailsReportImpl, GenerateDetailedCostReportImpl, GenerateDetailedCostReportOperationResultsImpl, GenerateDetailedCostReportOperationStatusImpl, PriceSheetImpl, ScheduledActionsImpl, BenefitRecommendationsImpl, BenefitUtilizationSummariesImpl } from "./operations";
export class CostManagementClient extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the CostManagementClient class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param options The parameter options
     */
    constructor(credentials, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-costmanagement/1.0.0-beta.1`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.apiVersion = options.apiVersion || "2022-10-01";
        this.operations = new OperationsImpl(this);
        this.views = new ViewsImpl(this);
        this.alerts = new AlertsImpl(this);
        this.forecast = new ForecastImpl(this);
        this.dimensions = new DimensionsImpl(this);
        this.query = new QueryImpl(this);
        this.generateReservationDetailsReport = new GenerateReservationDetailsReportImpl(this);
        this.exports = new ExportsImpl(this);
        this.generateCostDetailsReport = new GenerateCostDetailsReportImpl(this);
        this.generateDetailedCostReport = new GenerateDetailedCostReportImpl(this);
        this.generateDetailedCostReportOperationResults = new GenerateDetailedCostReportOperationResultsImpl(this);
        this.generateDetailedCostReportOperationStatus = new GenerateDetailedCostReportOperationStatusImpl(this);
        this.priceSheet = new PriceSheetImpl(this);
        this.scheduledActions = new ScheduledActionsImpl(this);
        this.benefitRecommendations = new BenefitRecommendationsImpl(this);
        this.benefitUtilizationSummaries = new BenefitUtilizationSummariesImpl(this);
        this.addCustomApiVersionPolicy(options.apiVersion);
    }
    /** A function that adds a policy that sets the api-version (or equivalent) to reflect the library version. */
    addCustomApiVersionPolicy(apiVersion) {
        if (!apiVersion) {
            return;
        }
        const apiVersionPolicy = {
            name: "CustomApiVersionPolicy",
            sendRequest(request, next) {
                return __awaiter(this, void 0, void 0, function* () {
                    const param = request.url.split("?");
                    if (param.length > 1) {
                        const newParams = param[1].split("&").map((item) => {
                            if (item.indexOf("api-version") > -1) {
                                return "api-version=" + apiVersion;
                            }
                            else {
                                return item;
                            }
                        });
                        request.url = param[0] + "?" + newParams.join("&");
                    }
                    return next(request);
                });
            }
        };
        this.pipeline.addPolicy(apiVersionPolicy);
    }
}
